<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Download and Explore Datasets from UCSC Xena Data Hubs • UCSCXenaTools</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Download and Explore Datasets from UCSC Xena Data Hubs">
<meta property="og:description" content="Download and explore datasets from UCSC Xena data hubs, which are
    a collection of UCSC-hosted public databases such as TCGA, ICGC, TARGET, GTEx, CCLE, and others.
    Databases are normalized so they can be combined, linked, filtered, explored and downloaded.
    This package is based on 'XenaR' package.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">UCSCXenaTools</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/USCSXenaTools_0.2.0.html">UCSCXenaTools: Download Public Cancer Genomic Data from UCSC Xena Hubs</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ShixiangWang/UCSCXenaTools">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    
<!-- README.md is generated from README.Rmd. Please edit that file -->
<div id="ucscxenatools-a-r-package-download-and-explore-data-from-ucsc-xena-data-hubs" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#ucscxenatools-a-r-package-download-and-explore-data-from-ucsc-xena-data-hubs" class="anchor"></a>UCSCXenaTools: A R package download and explore data from <strong>UCSC Xena data hubs</strong>
</h1></div>
<pre><code>![](http://www.r-pkg.org/badges/version-last-release/UCSCXenaTools) ![](http://cranlogs.r-pkg.org/badges/UCSCXenaTools?color=red)
[![Open Source Love svg1](https://badges.frapsoft.com/os/v1/open-source.svg?v=103)](https://github.com/ellerbrock/open-source-badges/) [![Maintenance](https://img.shields.io/badge/Maintained%3F-yes-green.svg)](https://GitHub.com/ShixiangWang/UCSCXenaTools/graphs/commit-activity) [![AppVeyor Build Status](https://ci.appveyor.com/api/projects/status/github/ShixiangWang/UCSCXenaTools?branch=master&amp;svg=true)](https://ci.appveyor.com/project/ShixiangWang/UCSCXenaTools) [![Coverage Status](https://img.shields.io/codecov/c/github/ShixiangWang/UCSCXenaTools/master.svg)](https://codecov.io/github/ShixiangWang/UCSCXenaTools?branch=master)</code></pre>
<p><strong>UCSC Xena data hubs</strong>, which are</p>
<blockquote>
<p>A collection of UCSC-hosted public databases such as TCGA, ICGC, TARGET, GTEx, CCLE, and others. Databases are normalized so they can be combined, linked, filtered, explored and downloaded.</p>
<p>– <a href="https://xena.ucsc.edu/">UCSC Xena</a></p>
</blockquote>
<p><strong>Current Version: 0.2.1</strong></p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install UCSCXenaTools from github with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install.packages("devtools")</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"ShixiangWang/UCSCXenaTools"</span>, <span class="dt">build_vignettes =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Read this vignettes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">browseVignettes</span>(<span class="st">"UCSCXenaTools"</span>)</code></pre></div>
</div>
<div id="data-hub-list" class="section level2">
<h2 class="hasAnchor">
<a href="#data-hub-list" class="anchor"></a>Data Hub List</h2>
<p>All datasets are available at <a href="https://xenabrowser.net/datapages/" class="uri">https://xenabrowser.net/datapages/</a>.</p>
<p>Currently, <code>UCSCXenaTools</code> support all 7 data hubs of UCSC Xena.</p>
<ul>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fucscpublic.xenahubs.net">UCSC Public Hub</a>: <a href="https://ucscpublic.xenahubs.net" class="uri">https://ucscpublic.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Ftcga.xenahubs.net">TCGA Hub</a>: <a href="https://tcga.xenahubs.net" class="uri">https://tcga.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fgdc.xenahubs.net">GDC Xena Hub</a>: <a href="https://gdc.xenahubs.net" class="uri">https://gdc.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Ficgc.xenahubs.net">ICGC Xena Hub</a>: <a href="https://icgc.xenahubs.net" class="uri">https://icgc.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fpancanatlas.xenahubs.net">Pan-Cancer Atlas Hub</a>: <a href="https://pancanatlas.xenahubs.net" class="uri">https://pancanatlas.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Ftoil.xenahubs.net">GA4GH (TOIL) Hub</a>: <a href="https://toil.xenahubs.net" class="uri">https://toil.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu">Treehouse Hub</a>: <a href="https://xena.treehouse.gi.ucsc.edu" class="uri">https://xena.treehouse.gi.ucsc.edu</a>
</li>
</ul>
<p>If the <code>API</code> changed, please remind me by email to <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a> or open an issue on <a href="https://github.com/ShixiangWang/UCSCXenaTools/issues">GitHub</a>.</p>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>Download UCSC Xena Datasets and load them into R by <code>UCSCXenaTools</code> is a workflow in <code>generate</code>, <code>filter</code>, <code>query</code>, <code>download</code> and <code>prepare</code> 5 steps, which are implemented as <code>XenaGenerate</code>, <code>XenaFilter</code>, <code>XenaQuery</code>, <code>XenaDownload</code> and <code>XenaPrepare</code>, respectively. They are very clear and easy to use and combine with other packages like <code>dplyr</code>.</p>
<p>The following use clinical data download of LUNG, LUAD, LUSC from TCGA (hg19 version) as an example.</p>
<div id="xenadata-data-frame" class="section level3">
<h3 class="hasAnchor">
<a href="#xenadata-data-frame" class="anchor"></a>XenaData data.frame</h3>
<p>Begin from version <code>0.2.0</code>, <code>UCSCXenaTools</code> use a <code>data.frame</code> object (built in package) <code>XenaData</code> to generate an instance of <code>XenaHub</code> class, which communicate with API of UCSC Xena Data Hubs.</p>
<p>You can load <code>XenaData</code> after loading <code>UCSCXenaTools</code> into R.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(UCSCXenaTools)
<span class="kw">data</span>(XenaData)

<span class="kw">head</span>(XenaData)
<span class="co">#&gt;                         XenaHosts XenaHostNames</span>
<span class="co">#&gt; 1 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 2 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 3 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 4 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 5 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 6 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt;                                     XenaCohorts</span>
<span class="co">#&gt; 1                                  1000_genomes</span>
<span class="co">#&gt; 2                                  1000_genomes</span>
<span class="co">#&gt; 3 Acute lymphoblastic leukemia (Mullighan 2008)</span>
<span class="co">#&gt; 4 Acute lymphoblastic leukemia (Mullighan 2008)</span>
<span class="co">#&gt; 5 Acute lymphoblastic leukemia (Mullighan 2008)</span>
<span class="co">#&gt; 6                          B cells (Basso 2005)</span>
<span class="co">#&gt;                                               XenaDatasets</span>
<span class="co">#&gt; 1                                       1000_genomes/BRCA2</span>
<span class="co">#&gt; 2                                       1000_genomes/BRCA1</span>
<span class="co">#&gt; 3    mullighan2008_public/mullighan2008_500K_genomicMatrix</span>
<span class="co">#&gt; 4 mullighan2008_public/mullighan2008_public_clinicalMatrix</span>
<span class="co">#&gt; 5    mullighan2008_public/mullighan2008_SNP6_genomicMatrix</span>
<span class="co">#&gt; 6         basso2005_public/basso2005_public_clinicalMatrix</span></code></pre></div>
</div>
<div id="generate-a-xenahub-object" class="section level3">
<h3 class="hasAnchor">
<a href="#generate-a-xenahub-object" class="anchor"></a>Generate a XenaHub object</h3>
<p>This can be implemented by <code>XenaGenerate</code> function, which generate <code>XenaHub</code> object from <code>XenaData</code> data frame.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/XenaGenerate.html">XenaGenerate</a></span>()
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://ucscpublic.xenahubs.net</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt;   https://gdc.xenahubs.net</span>
<span class="co">#&gt;   https://icgc.xenahubs.net</span>
<span class="co">#&gt;   https://toil.xenahubs.net</span>
<span class="co">#&gt; cohorts() (131 total):</span>
<span class="co">#&gt;   1000_genomes</span>
<span class="co">#&gt;   Acute lymphoblastic leukemia (Mullighan 2008)</span>
<span class="co">#&gt;   B cells (Basso 2005)</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA and TARGET Pan-Cancer (PANCAN)</span>
<span class="co">#&gt;   TCGA TARGET GTEx</span>
<span class="co">#&gt; datasets() (1521 total):</span>
<span class="co">#&gt;   1000_genomes/BRCA2</span>
<span class="co">#&gt;   1000_genomes/BRCA1</span>
<span class="co">#&gt;   mullighan2008_public/mullighan2008_500K_genomicMatrix</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TcgaTargetGtex_rsem_isoform_tpm</span>
<span class="co">#&gt;   TcgaTargetGtex_rsem_isopct</span></code></pre></div>
<p>We can set <code>subset</code> argument to narrow datasets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/XenaGenerate.html">XenaGenerate</a></span>(<span class="dt">subset =</span> XenaHostNames<span class="op">==</span><span class="st">"TCGA"</span>)
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt; cohorts() (38 total):</span>
<span class="co">#&gt;   TCGA Acute Myeloid Leukemia (LAML)</span>
<span class="co">#&gt;   TCGA Adrenocortical Cancer (ACC)</span>
<span class="co">#&gt;   TCGA Bile Duct Cancer (CHOL)</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA Thyroid Cancer (THCA)</span>
<span class="co">#&gt;   TCGA Uterine Carcinosarcoma (UCS)</span>
<span class="co">#&gt; datasets() (879 total):</span>
<span class="co">#&gt;   TCGA.LAML.sampleMap/HumanMethylation27</span>
<span class="co">#&gt;   TCGA.LAML.sampleMap/HumanMethylation450</span>
<span class="co">#&gt;   TCGA.LAML.sampleMap/Gistic2_CopyNumber_Gistic2_all_data_by_genes</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA.UCS.sampleMap/Pathway_Paradigm_RNASeq_And_Copy_Number</span>
<span class="co">#&gt;   TCGA.UCS.sampleMap/mutation_curated_broad</span></code></pre></div>
<blockquote>
<p>You can use <code><a href="reference/XenaHub.html">XenaHub()</a></code> to generate a <code>XenaHub</code> object for API communication, but it is not recommended.</p>
</blockquote>
<p>It’s possible to explore <code><a href="reference/hosts.html">hosts()</a></code>, <code><a href="reference/cohorts.html">cohorts()</a></code> and <code><a href="reference/datasets.html">datasets()</a></code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">xe =<span class="st"> </span><span class="kw"><a href="reference/XenaGenerate.html">XenaGenerate</a></span>(<span class="dt">subset =</span> XenaHostNames<span class="op">==</span><span class="st">"TCGA"</span>)
<span class="co"># get hosts</span>
<span class="kw"><a href="reference/hosts.html">hosts</a></span>(xe)
<span class="co">#&gt; [1] "https://tcga.xenahubs.net"</span>
<span class="co"># get cohorts</span>
<span class="kw">head</span>(<span class="kw"><a href="reference/cohorts.html">cohorts</a></span>(xe))
<span class="co">#&gt; [1] "TCGA Acute Myeloid Leukemia (LAML)"</span>
<span class="co">#&gt; [2] "TCGA Adrenocortical Cancer (ACC)"  </span>
<span class="co">#&gt; [3] "TCGA Bile Duct Cancer (CHOL)"      </span>
<span class="co">#&gt; [4] "TCGA Bladder Cancer (BLCA)"        </span>
<span class="co">#&gt; [5] "TCGA Breast Cancer (BRCA)"         </span>
<span class="co">#&gt; [6] "TCGA Cervical Cancer (CESC)"</span>
<span class="co"># get datasets</span>
<span class="kw">head</span>(<span class="kw"><a href="reference/datasets.html">datasets</a></span>(xe))
<span class="co">#&gt; [1] "TCGA.LAML.sampleMap/HumanMethylation27"                          </span>
<span class="co">#&gt; [2] "TCGA.LAML.sampleMap/HumanMethylation450"                         </span>
<span class="co">#&gt; [3] "TCGA.LAML.sampleMap/Gistic2_CopyNumber_Gistic2_all_data_by_genes"</span>
<span class="co">#&gt; [4] "TCGA.LAML.sampleMap/mutation_wustl_hiseq"                        </span>
<span class="co">#&gt; [5] "TCGA.LAML.sampleMap/GA"                                          </span>
<span class="co">#&gt; [6] "TCGA.LAML.sampleMap/HiSeqV2_percentile"</span></code></pre></div>
<p>Pipe operator <code>%&gt;%</code> can also be used here.</p>
<pre><code>&gt; library(tidyverse)
&gt; XenaData %&gt;% filter(XenaHostNames == "TCGA", grepl("BRCA", XenaCohorts), grepl("Path", XenaDatasets)) %&gt;% XenaGenerate()
class: XenaHub 
hosts():
  https://tcga.xenahubs.net
cohorts() (1 total):
  TCGA Breast Cancer (BRCA)
datasets() (4 total):
  TCGA.BRCA.sampleMap/Pathway_Paradigm_mRNA_And_Copy_Number
  TCGA.BRCA.sampleMap/Pathway_Paradigm_RNASeq
  TCGA.BRCA.sampleMap/Pathway_Paradigm_RNASeq_And_Copy_Number
  TCGA.BRCA.sampleMap/Pathway_Paradigm_mRNA</code></pre>
</div>
<div id="filter" class="section level3">
<h3 class="hasAnchor">
<a href="#filter" class="anchor"></a>Filter</h3>
<p>There are too many datasets, we filter them by <code>XenaFilter</code> function.</p>
<p>Regular expression can be used to filter XenaHub object to what we want.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(<span class="kw"><a href="reference/XenaFilter.html">XenaFilter</a></span>(xe, <span class="dt">filterDatasets =</span> <span class="st">"clinical"</span>) -&gt;<span class="st"> </span>xe2)
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt; cohorts() (39 total):</span>
<span class="co">#&gt;   (unassigned)</span>
<span class="co">#&gt;   TCGA Acute Myeloid Leukemia (LAML)</span>
<span class="co">#&gt;   TCGA Adrenocortical Cancer (ACC)</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA Thyroid Cancer (THCA)</span>
<span class="co">#&gt;   TCGA Uterine Carcinosarcoma (UCS)</span>
<span class="co">#&gt; datasets() (37 total):</span>
<span class="co">#&gt;   TCGA.OV.sampleMap/OV_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.DLBC.sampleMap/DLBC_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.KIRC.sampleMap/KIRC_clinicalMatrix</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA.READ.sampleMap/READ_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.MESO.sampleMap/MESO_clinicalMatrix</span></code></pre></div>
<p>Then select <code>LUAD</code>, <code>LUSC</code> and <code>LUNG</code> 3 datasets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/XenaFilter.html">XenaFilter</a></span>(xe2, <span class="dt">filterDatasets =</span> <span class="st">"LUAD|LUSC|LUNG"</span>) -&gt;<span class="st"> </span>xe2</code></pre></div>
<p>Pipe can be used here.</p>
<pre><code>suppressMessages(require(dplyr))

xe %&gt;% 
    filterXena(filterDatasets = "clinical") %&gt;% 
    filterXena(filterDatasets = "luad|lusc|lung")
## class: XenaHub 
## hosts():
##   https://tcga.xenahubs.net
## cohorts() (39 total):
##   (unassigned)
##   TCGA Acute Myeloid Leukemia (LAML)
##   TCGA Adrenocortical Cancer (ACC)
##   ...
##   TCGA Thyroid Cancer (THCA)
##   TCGA Uterine Carcinosarcoma (UCS)
## datasets() (3 total):
##   TCGA.LUSC.sampleMap/LUSC_clinicalMatrix
##   TCGA.LUNG.sampleMap/LUNG_clinicalMatrix
##   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</code></pre>
</div>
<div id="query" class="section level3">
<h3 class="hasAnchor">
<a href="#query" class="anchor"></a>Query</h3>
<p>Create a query before download data</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">xe2_query =<span class="st"> </span><span class="kw"><a href="reference/XenaQuery.html">XenaQuery</a></span>(xe2)
xe2_query
<span class="co">#&gt;                       hosts                                datasets</span>
<span class="co">#&gt; 1 https://tcga.xenahubs.net TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span>
<span class="co">#&gt; 2 https://tcga.xenahubs.net TCGA.LUNG.sampleMap/LUNG_clinicalMatrix</span>
<span class="co">#&gt; 3 https://tcga.xenahubs.net TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span>
<span class="co">#&gt;                                                                             url</span>
<span class="co">#&gt; 1 https://tcga.xenahubs.net/download/TCGA.LUSC.sampleMap/LUSC_clinicalMatrix.gz</span>
<span class="co">#&gt; 2 https://tcga.xenahubs.net/download/TCGA.LUNG.sampleMap/LUNG_clinicalMatrix.gz</span>
<span class="co">#&gt; 3 https://tcga.xenahubs.net/download/TCGA.LUAD.sampleMap/LUAD_clinicalMatrix.gz</span></code></pre></div>
</div>
<div id="download" class="section level3">
<h3 class="hasAnchor">
<a href="#download" class="anchor"></a>Download</h3>
<p>Default, data will be downloaded to <code>XenaData</code> directory under system temp directory. You can specify the path.</p>
<p>If the data exists, command will not run to download them, but you can force it by <code>force</code> option.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">xe2_download =<span class="st"> </span><span class="kw"><a href="reference/XenaDownload.html">XenaDownload</a></span>(xe2_query)
<span class="co">#&gt; We will download files to directory D:\tmp\Rtmp67d6MOXenaData.</span>
<span class="co">#&gt; Downloading TCGA.LUSC.sampleMap__LUSC_clinicalMatrix.gz</span>
<span class="co">#&gt; Downloading TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz</span>
<span class="co">#&gt; Downloading TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz</span>
<span class="co">#&gt; Note fileNames transfromed from datasets name and / chracter all changed to __ character.</span>
## not run
<span class="co">#xe2_download = XenaDownload(xe2_query, destdir = "E:/Github/XenaData/test/")</span></code></pre></div>
<blockquote>
<p>Note fileNames transfromed from datasets name and / chracter all changed to __ character.</p>
</blockquote>
</div>
<div id="prepare" class="section level3">
<h3 class="hasAnchor">
<a href="#prepare" class="anchor"></a>Prepare</h3>
<p>There are 4 ways to prepare data to R.</p>
<pre><code># way1:  directory
cli1 = XenaPrepare("E:/Github/XenaData/test/")
names(cli1)
## [1] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"
## [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"
## [3] "TCGA.LUSC.sampleMap__LUSC_clinicalMatrix.gz"</code></pre>
<pre><code># way2: local files
cli2 = XenaPrepare("E:/Github/XenaData/test/TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz")
class(cli2)
## [1] "tbl_df"     "tbl"        "data.frame"

cli2 = XenaPrepare(c("E:/Github/XenaData/test/TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz",
                     "E:/Github/XenaData/test/TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"))
class(cli2)
## [1] "list"
names(cli2)
## [1] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"
## [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"</code></pre>
<pre><code># way3: urls
cli3 = XenaPrepare(xe2_download$url[1:2])
names(cli3)
## [1] "LUSC_clinicalMatrix.gz" "LUNG_clinicalMatrix.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># way4: xenadownload object</span>
cli4 =<span class="st"> </span><span class="kw"><a href="reference/XenaPrepare.html">XenaPrepare</a></span>(xe2_download)
<span class="kw">names</span>(cli4)
<span class="co">#&gt; [1] "TCGA.LUSC.sampleMap__LUSC_clinicalMatrix.gz"</span>
<span class="co">#&gt; [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"</span>
<span class="co">#&gt; [3] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"</span></code></pre></div>
</div>
<div id="sessioninfo" class="section level3">
<h3 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.5.1 (2018-07-02)</span>
<span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">#&gt; Running under: Windows 7 x64 (build 7601) Service Pack 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] LC_COLLATE=Chinese (Simplified)_People's Republic of China.936 </span>
<span class="co">#&gt; [2] LC_CTYPE=Chinese (Simplified)_People's Republic of China.936   </span>
<span class="co">#&gt; [3] LC_MONETARY=Chinese (Simplified)_People's Republic of China.936</span>
<span class="co">#&gt; [4] LC_NUMERIC=C                                                   </span>
<span class="co">#&gt; [5] LC_TIME=Chinese (Simplified)_People's Republic of China.936    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] UCSCXenaTools_0.2.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_0.12.18       rstudioapi_0.7     knitr_1.20        </span>
<span class="co">#&gt;  [4] xml2_1.2.0         magrittr_1.5       roxygen2_6.1.0    </span>
<span class="co">#&gt;  [7] hms_0.4.2          MASS_7.3-50        R6_2.2.2          </span>
<span class="co">#&gt; [10] rlang_0.2.1        httr_1.3.1         stringr_1.3.1     </span>
<span class="co">#&gt; [13] tools_3.5.1        htmltools_0.3.6    commonmark_1.5    </span>
<span class="co">#&gt; [16] yaml_2.2.0         rprojroot_1.3-2    digest_0.6.15     </span>
<span class="co">#&gt; [19] assertthat_0.2.0   tibble_1.4.2       pkgdown_1.1.0.9000</span>
<span class="co">#&gt; [22] crayon_1.3.4       readr_1.1.1        fs_1.2.5          </span>
<span class="co">#&gt; [25] curl_3.2           memoise_1.1.0      evaluate_0.11     </span>
<span class="co">#&gt; [28] rmarkdown_1.10     stringi_1.1.7      pillar_1.3.0      </span>
<span class="co">#&gt; [31] compiler_3.5.1     desc_1.2.0         backports_1.1.2   </span>
<span class="co">#&gt; [34] jsonlite_1.5       pkgconfig_2.0.1</span></code></pre></div>
</div>
</div>
<div id="acknowledgement" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgement" class="anchor"></a>Acknowledgement</h2>
<p>This package is based on <a href="https://github.com/mtmorgan/XenaR">XenaR</a>, thanks <a href="https://github.com/mtmorgan">Martin Morgan</a> for his work.</p>
</div>
<div id="license" class="section level2">
<h2 class="hasAnchor">
<a href="#license" class="anchor"></a>LICENSE</h2>
<p>GPL-3</p>
<p>please note, code from XenaR package under Apache 2.0 license.</p>
</div>
<div id="todo" class="section level2">
<h2 class="hasAnchor">
<a href="#todo" class="anchor"></a>ToDo</h2>
<ul>
<li>Shinny</li>
<li>More easier download workflow</li>
</ul>
</div>
<div id="code-of-conduct" class="section level2">
<h2 class="hasAnchor">
<a href="#code-of-conduct" class="anchor"></a>Code of conduct</h2>
<p>Please note that this project is released with a <a href="CODE_OF_CONDUCT.html">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/ShixiangWang/UCSCXenaTools">https://​github.com/​ShixiangWang/​UCSCXenaTools</a>
</li>
<li>Report a bug at <br><a href="https://github.com/ShixiangWang/UCSCXenaTools/issues">https://​github.com/​ShixiangWang/​UCSCXenaTools/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Shixiang Wang <br><small class="roles"> Author, maintainer </small>  </li>
<li>Martin Morgan <br><small class="roles"> Author </small>  </li>
</ul>
</div>

      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Shixiang Wang, Martin Morgan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
