<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>UCSCXenaTools: Download Public Cancer Genomic Data from UCSC Xena Hubs • UCSCXenaTools</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="UCSCXenaTools: Download Public Cancer Genomic Data from UCSC Xena Hubs">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">UCSCXenaTools</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.2.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/USCSXenaTools.html">UCSCXenaTools: Download Public Cancer Genomic Data from UCSC Xena Hubs</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ShixiangWang/UCSCXenaTools">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>UCSCXenaTools: Download Public Cancer Genomic Data from UCSC Xena Hubs</h1>
                        <h4 class="author">Shixiang Wang<br>
ShanghaiTech. University</h4>
            
            <h4 class="date">2018-09-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ShixiangWang/UCSCXenaTools/blob/master/vignettes/USCSXenaTools.Rmd"><code>vignettes/USCSXenaTools.Rmd</code></a></small>
      <div class="hidden name"><code>USCSXenaTools.Rmd</code></div>

    </div>

    
    
<p><strong>Current Version: 0.2.4</strong></p>
<p><code>UCSCXenaTools</code> is a R package download and explore data from <strong>UCSC Xena data hubs</strong>, which are</p>
<blockquote>
<p>A collection of UCSC-hosted public databases such as TCGA, ICGC, TARGET, GTEx, CCLE, and others. Databases are normalized so they can be combined, linked, filtered, explored and downloaded.</p>
<p>– <a href="https://xena.ucsc.edu/">UCSC Xena</a></p>
</blockquote>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Install stable release from CRAN with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">"UCSCXenaTools"</span>)</code></pre></div>
<p>You can also install devel version of UCSCXenaTools from github with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install.packages("devtools")</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"ShixiangWang/UCSCXenaTools"</span>, <span class="dt">build_vignettes =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Read this vignettes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">browseVignettes</span>(<span class="st">"UCSCXenaTools"</span>)
<span class="co"># or</span>
??UCSCXenaTools</code></pre></div>
</div>
<div id="data-hub-list" class="section level2">
<h2 class="hasAnchor">
<a href="#data-hub-list" class="anchor"></a>Data Hub List</h2>
<p>All datasets are available at <a href="https://xenabrowser.net/datapages/" class="uri">https://xenabrowser.net/datapages/</a>.</p>
<p>Currently, <code>UCSCXenaTools</code> support all 7 data hubs of UCSC Xena.</p>
<ul>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fucscpublic.xenahubs.net">UCSC Public Hub</a>: <a href="https://ucscpublic.xenahubs.net" class="uri">https://ucscpublic.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Ftcga.xenahubs.net">TCGA Hub</a>: <a href="https://tcga.xenahubs.net" class="uri">https://tcga.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fgdc.xenahubs.net">GDC Xena Hub</a>: <a href="https://gdc.xenahubs.net" class="uri">https://gdc.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Ficgc.xenahubs.net">ICGC Xena Hub</a>: <a href="https://icgc.xenahubs.net" class="uri">https://icgc.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fpancanatlas.xenahubs.net">Pan-Cancer Atlas Hub</a>: <a href="https://pancanatlas.xenahubs.net" class="uri">https://pancanatlas.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Ftoil.xenahubs.net">GA4GH (TOIL) Hub</a>: <a href="https://toil.xenahubs.net" class="uri">https://toil.xenahubs.net</a>
</li>
<li>
<a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu">Treehouse Hub</a>: <a href="https://xena.treehouse.gi.ucsc.edu" class="uri">https://xena.treehouse.gi.ucsc.edu</a>
</li>
</ul>
<p>If the <code>API</code> changed, please remind me by email to <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a> or open an issue on <a href="https://github.com/ShixiangWang/UCSCXenaTools/issues">GitHub</a>.</p>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>Download UCSC Xena Datasets and load them into R by <code>UCSCXenaTools</code> is a workflow in <code>generate</code>, <code>filter</code>, <code>query</code>, <code>download</code> and <code>prepare</code> 5 steps, which are implemented as <code>XenaGenerate</code>, <code>XenaFilter</code>, <code>XenaQuery</code>, <code>XenaDownload</code> and <code>XenaPrepare</code>, respectively. They are very clear and easy to use and combine with other packages like <code>dplyr</code>.</p>
<p>The following use clinical data download of LUNG, LUAD, LUSC from TCGA (hg19 version) as an example.</p>
<div id="xenadata-data-frame" class="section level3">
<h3 class="hasAnchor">
<a href="#xenadata-data-frame" class="anchor"></a>XenaData data.frame</h3>
<p>Begin from version <code>0.2.0</code>, <code>UCSCXenaTools</code> use a <code>data.frame</code> object (built in package) <code>XenaData</code> to generate an instance of <code>XenaHub</code> class, which communicate with API of UCSC Xena Data Hubs.</p>
<p>You can load <code>XenaData</code> after loading <code>UCSCXenaTools</code> into R.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(UCSCXenaTools)
<span class="kw">data</span>(XenaData)

<span class="kw">head</span>(XenaData)
<span class="co">#&gt;                         XenaHosts XenaHostNames</span>
<span class="co">#&gt; 1 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 2 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 3 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 4 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 5 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt; 6 https://ucscpublic.xenahubs.net   UCSC_Public</span>
<span class="co">#&gt;                                     XenaCohorts</span>
<span class="co">#&gt; 1                                  1000_genomes</span>
<span class="co">#&gt; 2                                  1000_genomes</span>
<span class="co">#&gt; 3 Acute lymphoblastic leukemia (Mullighan 2008)</span>
<span class="co">#&gt; 4 Acute lymphoblastic leukemia (Mullighan 2008)</span>
<span class="co">#&gt; 5 Acute lymphoblastic leukemia (Mullighan 2008)</span>
<span class="co">#&gt; 6                          B cells (Basso 2005)</span>
<span class="co">#&gt;                                               XenaDatasets</span>
<span class="co">#&gt; 1                                       1000_genomes/BRCA2</span>
<span class="co">#&gt; 2                                       1000_genomes/BRCA1</span>
<span class="co">#&gt; 3    mullighan2008_public/mullighan2008_500K_genomicMatrix</span>
<span class="co">#&gt; 4 mullighan2008_public/mullighan2008_public_clinicalMatrix</span>
<span class="co">#&gt; 5    mullighan2008_public/mullighan2008_SNP6_genomicMatrix</span>
<span class="co">#&gt; 6         basso2005_public/basso2005_public_clinicalMatrix</span></code></pre></div>
</div>
<div id="generate-a-xenahub-object" class="section level3">
<h3 class="hasAnchor">
<a href="#generate-a-xenahub-object" class="anchor"></a>Generate a XenaHub object</h3>
<p>This can be implemented by <code>XenaGenerate</code> function, which generate <code>XenaHub</code> object from <code>XenaData</code> data frame.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span>()
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://ucscpublic.xenahubs.net</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt;   https://gdc.xenahubs.net</span>
<span class="co">#&gt;   https://icgc.xenahubs.net</span>
<span class="co">#&gt;   https://toil.xenahubs.net</span>
<span class="co">#&gt;   https://pancanatlas.xenahubs.net</span>
<span class="co">#&gt;   https://xena.treehouse.gi.ucsc.edu</span>
<span class="co">#&gt; cohorts() (134 total):</span>
<span class="co">#&gt;   1000_genomes</span>
<span class="co">#&gt;   Acute lymphoblastic leukemia (Mullighan 2008)</span>
<span class="co">#&gt;   B cells (Basso 2005)</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   Treehouse PED v8</span>
<span class="co">#&gt;   Treehouse public expression dataset (July 2017)</span>
<span class="co">#&gt; datasets() (1549 total):</span>
<span class="co">#&gt;   1000_genomes/BRCA2</span>
<span class="co">#&gt;   1000_genomes/BRCA1</span>
<span class="co">#&gt;   mullighan2008_public/mullighan2008_500K_genomicMatrix</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   treehouse_public_samples_unique_ensembl_expected_count.2017-09-11.tsv</span>
<span class="co">#&gt;   treehouse_public_samples_unique_hugo_log2_tpm_plus_1.2017-09-11.tsv</span></code></pre></div>
<p>We can set <code>subset</code> argument to narrow datasets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span>(<span class="dt">subset =</span> XenaHostNames<span class="op">==</span><span class="st">"TCGA"</span>)
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt; cohorts() (38 total):</span>
<span class="co">#&gt;   TCGA Acute Myeloid Leukemia (LAML)</span>
<span class="co">#&gt;   TCGA Adrenocortical Cancer (ACC)</span>
<span class="co">#&gt;   TCGA Bile Duct Cancer (CHOL)</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA Thyroid Cancer (THCA)</span>
<span class="co">#&gt;   TCGA Uterine Carcinosarcoma (UCS)</span>
<span class="co">#&gt; datasets() (879 total):</span>
<span class="co">#&gt;   TCGA.LAML.sampleMap/HumanMethylation27</span>
<span class="co">#&gt;   TCGA.LAML.sampleMap/HumanMethylation450</span>
<span class="co">#&gt;   TCGA.LAML.sampleMap/Gistic2_CopyNumber_Gistic2_all_data_by_genes</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA.UCS.sampleMap/Pathway_Paradigm_RNASeq_And_Copy_Number</span>
<span class="co">#&gt;   TCGA.UCS.sampleMap/mutation_curated_broad</span></code></pre></div>
<blockquote>
<p>You can use <code><a href="../reference/XenaHub.html">XenaHub()</a></code> to generate a <code>XenaHub</code> object for API communication, but it is not recommended.</p>
</blockquote>
<p>It’s possible to explore <code><a href="../reference/hosts.html">hosts()</a></code>, <code><a href="../reference/cohorts.html">cohorts()</a></code> and <code><a href="../reference/datasets.html">datasets()</a></code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">xe =<span class="st"> </span><span class="kw"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span>(<span class="dt">subset =</span> XenaHostNames<span class="op">==</span><span class="st">"TCGA"</span>)
<span class="co"># get hosts</span>
<span class="kw"><a href="../reference/hosts.html">hosts</a></span>(xe)
<span class="co">#&gt; [1] "https://tcga.xenahubs.net"</span>
<span class="co"># get cohorts</span>
<span class="kw">head</span>(<span class="kw"><a href="../reference/cohorts.html">cohorts</a></span>(xe))
<span class="co">#&gt; [1] "TCGA Acute Myeloid Leukemia (LAML)"</span>
<span class="co">#&gt; [2] "TCGA Adrenocortical Cancer (ACC)"  </span>
<span class="co">#&gt; [3] "TCGA Bile Duct Cancer (CHOL)"      </span>
<span class="co">#&gt; [4] "TCGA Bladder Cancer (BLCA)"        </span>
<span class="co">#&gt; [5] "TCGA Breast Cancer (BRCA)"         </span>
<span class="co">#&gt; [6] "TCGA Cervical Cancer (CESC)"</span>
<span class="co"># get datasets</span>
<span class="kw">head</span>(<span class="kw"><a href="../reference/datasets.html">datasets</a></span>(xe))
<span class="co">#&gt; [1] "TCGA.LAML.sampleMap/HumanMethylation27"                          </span>
<span class="co">#&gt; [2] "TCGA.LAML.sampleMap/HumanMethylation450"                         </span>
<span class="co">#&gt; [3] "TCGA.LAML.sampleMap/Gistic2_CopyNumber_Gistic2_all_data_by_genes"</span>
<span class="co">#&gt; [4] "TCGA.LAML.sampleMap/mutation_wustl_hiseq"                        </span>
<span class="co">#&gt; [5] "TCGA.LAML.sampleMap/GA"                                          </span>
<span class="co">#&gt; [6] "TCGA.LAML.sampleMap/HiSeqV2_percentile"</span></code></pre></div>
<p>Pipe operator <code>%&gt;%</code> can also be used here.</p>
<pre><code>&gt; library(tidyverse)
&gt; XenaData %&gt;% filter(XenaHostNames == "TCGA", grepl("BRCA", XenaCohorts), grepl("Path", XenaDatasets)) %&gt;% XenaGenerate()
class: XenaHub 
hosts():
  https://tcga.xenahubs.net
cohorts() (1 total):
  TCGA Breast Cancer (BRCA)
datasets() (4 total):
  TCGA.BRCA.sampleMap/Pathway_Paradigm_mRNA_And_Copy_Number
  TCGA.BRCA.sampleMap/Pathway_Paradigm_RNASeq
  TCGA.BRCA.sampleMap/Pathway_Paradigm_RNASeq_And_Copy_Number
  TCGA.BRCA.sampleMap/Pathway_Paradigm_mRNA</code></pre>
</div>
<div id="filter" class="section level3">
<h3 class="hasAnchor">
<a href="#filter" class="anchor"></a>Filter</h3>
<p>There are too many datasets, we filter them by <code>XenaFilter</code> function.</p>
<p>Regular expression can be used to filter XenaHub object to what we want.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(<span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(xe, <span class="dt">filterDatasets =</span> <span class="st">"clinical"</span>) -&gt;<span class="st"> </span>xe2)
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt; cohorts() (39 total):</span>
<span class="co">#&gt;   (unassigned)</span>
<span class="co">#&gt;   TCGA Acute Myeloid Leukemia (LAML)</span>
<span class="co">#&gt;   TCGA Adrenocortical Cancer (ACC)</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA Thyroid Cancer (THCA)</span>
<span class="co">#&gt;   TCGA Uterine Carcinosarcoma (UCS)</span>
<span class="co">#&gt; datasets() (37 total):</span>
<span class="co">#&gt;   TCGA.OV.sampleMap/OV_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.DLBC.sampleMap/DLBC_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.KIRC.sampleMap/KIRC_clinicalMatrix</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA.READ.sampleMap/READ_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.MESO.sampleMap/MESO_clinicalMatrix</span></code></pre></div>
<p>Then select <code>LUAD</code>, <code>LUSC</code> and <code>LUNG</code> 3 datasets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(xe2, <span class="dt">filterDatasets =</span> <span class="st">"LUAD|LUSC|LUNG"</span>) -&gt;<span class="st"> </span>xe2</code></pre></div>
<p>Pipe can be used here.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressMessages</span>(<span class="kw">require</span>(dplyr))

xe <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(<span class="dt">filterDatasets =</span> <span class="st">"clinical"</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw"><a href="../reference/XenaFilter.html">XenaFilter</a></span>(<span class="dt">filterDatasets =</span> <span class="st">"luad|lusc|lung"</span>)
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt; cohorts() (39 total):</span>
<span class="co">#&gt;   (unassigned)</span>
<span class="co">#&gt;   TCGA Acute Myeloid Leukemia (LAML)</span>
<span class="co">#&gt;   TCGA Adrenocortical Cancer (ACC)</span>
<span class="co">#&gt;   ...</span>
<span class="co">#&gt;   TCGA Thyroid Cancer (THCA)</span>
<span class="co">#&gt;   TCGA Uterine Carcinosarcoma (UCS)</span>
<span class="co">#&gt; datasets() (3 total):</span>
<span class="co">#&gt;   TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.LUNG.sampleMap/LUNG_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span></code></pre></div>
</div>
<div id="query" class="section level3">
<h3 class="hasAnchor">
<a href="#query" class="anchor"></a>Query</h3>
<p>Create a query before download data</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">xe2_query =<span class="st"> </span><span class="kw"><a href="../reference/XenaQuery.html">XenaQuery</a></span>(xe2)
xe2_query
<span class="co">#&gt;                       hosts                                datasets</span>
<span class="co">#&gt; 1 https://tcga.xenahubs.net TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span>
<span class="co">#&gt; 2 https://tcga.xenahubs.net TCGA.LUNG.sampleMap/LUNG_clinicalMatrix</span>
<span class="co">#&gt; 3 https://tcga.xenahubs.net TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span>
<span class="co">#&gt;                                                                             url</span>
<span class="co">#&gt; 1 https://tcga.xenahubs.net/download/TCGA.LUSC.sampleMap/LUSC_clinicalMatrix.gz</span>
<span class="co">#&gt; 2 https://tcga.xenahubs.net/download/TCGA.LUNG.sampleMap/LUNG_clinicalMatrix.gz</span>
<span class="co">#&gt; 3 https://tcga.xenahubs.net/download/TCGA.LUAD.sampleMap/LUAD_clinicalMatrix.gz</span></code></pre></div>
</div>
<div id="download" class="section level3">
<h3 class="hasAnchor">
<a href="#download" class="anchor"></a>Download</h3>
<p>Default, data will be downloaded to system temp directory. You can specify the path.</p>
<p>If the data exists, command will not run to download them, but you can force it by <code>force</code> option.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">xe2_download =<span class="st"> </span><span class="kw"><a href="../reference/XenaDownload.html">XenaDownload</a></span>(xe2_query)
<span class="co">#&gt; We will download files to directory /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//Rtmpw9sHFT.</span>
<span class="co">#&gt; Downloading TCGA.LUSC.sampleMap__LUSC_clinicalMatrix.gz</span>
<span class="co">#&gt; Downloading TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz</span>
<span class="co">#&gt; Downloading TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz</span>
<span class="co">#&gt; Note fileNames transfromed from datasets name and / chracter all changed to __ character.</span>
## not run
<span class="co">#xe2_download = XenaDownload(xe2_query, destdir = "E:/Github/XenaData/test/")</span></code></pre></div>
<blockquote>
<p>Note fileNames transfromed from datasets name and / chracter all changed to __ character.</p>
</blockquote>
</div>
<div id="prepare" class="section level3">
<h3 class="hasAnchor">
<a href="#prepare" class="anchor"></a>Prepare</h3>
<p>There are 4 ways to prepare data to R.</p>
<pre><code># way1:  directory
cli1 = XenaPrepare("E:/Github/XenaData/test/")
names(cli1)
## [1] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"
## [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"
## [3] "TCGA.LUSC.sampleMap__LUSC_clinicalMatrix.gz"</code></pre>
<pre><code># way2: local files
cli2 = XenaPrepare("E:/Github/XenaData/test/TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz")
class(cli2)
## [1] "tbl_df"     "tbl"        "data.frame"

cli2 = XenaPrepare(c("E:/Github/XenaData/test/TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz",
                     "E:/Github/XenaData/test/TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"))
class(cli2)
## [1] "list"
names(cli2)
## [1] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"
## [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"</code></pre>
<pre><code># way3: urls
cli3 = XenaPrepare(xe2_download$url[1:2])
names(cli3)
## [1] "LUSC_clinicalMatrix.gz" "LUNG_clinicalMatrix.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># way4: xenadownload object</span>
cli4 =<span class="st"> </span><span class="kw"><a href="../reference/XenaPrepare.html">XenaPrepare</a></span>(xe2_download)
<span class="kw">names</span>(cli4)
<span class="co">#&gt; [1] "TCGA.LUSC.sampleMap__LUSC_clinicalMatrix.gz"</span>
<span class="co">#&gt; [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"</span>
<span class="co">#&gt; [3] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"</span></code></pre></div>
</div>
</div>
<div id="tcga-common-data-easy-download" class="section level2">
<h2 class="hasAnchor">
<a href="#tcga-common-data-easy-download" class="anchor"></a>TCGA Common Data Easy Download</h2>
<div id="gettcgadata" class="section level3">
<h3 class="hasAnchor">
<a href="#gettcgadata" class="anchor"></a>getTCGAdata</h3>
<p><code>getTCGAdata</code> provide a quite easy download way for TCGA datasets, user can specify multiple options to select which data and corresponding file type want to download. Default this function will return a list include <code>XenaHub</code> object and selective datasets information. Once you are sure the datasets is exactly what you want, <code>download</code> can be set to <code>TRUE</code> to download the data.</p>
<p>Check arguments of <code>getTCGAdata</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">args</span>(getTCGAdata)
<span class="co">#&gt; function (project = NULL, clinical = TRUE, download = FALSE, </span>
<span class="co">#&gt;     forceDownload = FALSE, destdir = tempdir(), mRNASeq = FALSE, </span>
<span class="co">#&gt;     mRNAArray = FALSE, mRNASeqType = c("normalized", "pancan normalized", </span>
<span class="co">#&gt;         "percentile"), miRNASeq = FALSE, exonRNASeq = FALSE, </span>
<span class="co">#&gt;     RPPAArray = FALSE, ReplicateBaseNormalization = FALSE, Methylation = FALSE, </span>
<span class="co">#&gt;     MethylationType = c("27K", "450K"), GeneMutation = FALSE, </span>
<span class="co">#&gt;     SomaticMutation = FALSE, GisticCopyNumber = FALSE, Gistic2Threshold = TRUE, </span>
<span class="co">#&gt;     CopyNumberSegment = FALSE, RemoveGermlineCNV = TRUE, ...) </span>
<span class="co">#&gt; NULL</span>

<span class="co"># or run</span>
<span class="co"># ??getTCGAdata to read documentation</span></code></pre></div>
<p>Select one or more projects, default will select only clinical datasets:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getTCGAdata.html">getTCGAdata</a></span>(<span class="kw">c</span>(<span class="st">"UVM"</span>, <span class="st">"LUAD"</span>))
<span class="co">#&gt; $Xena</span>
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt; cohorts() (2 total):</span>
<span class="co">#&gt;   TCGA Lung Adenocarcinoma (LUAD)</span>
<span class="co">#&gt;   TCGA Ocular melanomas (UVM)</span>
<span class="co">#&gt; datasets() (2 total):</span>
<span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.UVM.sampleMap/UVM_clinicalMatrix</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $DataInfo</span>
<span class="co">#&gt;                   XenaHosts XenaHostNames                     XenaCohorts</span>
<span class="co">#&gt; 1 https://tcga.xenahubs.net          TCGA TCGA Lung Adenocarcinoma (LUAD)</span>
<span class="co">#&gt; 2 https://tcga.xenahubs.net          TCGA     TCGA Ocular melanomas (UVM)</span>
<span class="co">#&gt;                              XenaDatasets ProjectID  DataType</span>
<span class="co">#&gt; 1 TCGA.LUAD.sampleMap/LUAD_clinicalMatrix      LUAD Phenotype</span>
<span class="co">#&gt; 2   TCGA.UVM.sampleMap/UVM_clinicalMatrix       UVM Phenotype</span>
<span class="co">#&gt;               FileType</span>
<span class="co">#&gt; 1 Clinical Information</span>
<span class="co">#&gt; 2 Clinical Information</span>

tcga_data =<span class="st"> </span><span class="kw"><a href="../reference/getTCGAdata.html">getTCGAdata</a></span>(<span class="kw">c</span>(<span class="st">"UVM"</span>, <span class="st">"LUAD"</span>))

<span class="co"># only return XenaHub object</span>
tcga_data<span class="op">$</span>Xena
<span class="co">#&gt; class: XenaHub </span>
<span class="co">#&gt; hosts():</span>
<span class="co">#&gt;   https://tcga.xenahubs.net</span>
<span class="co">#&gt; cohorts() (2 total):</span>
<span class="co">#&gt;   TCGA Lung Adenocarcinoma (LUAD)</span>
<span class="co">#&gt;   TCGA Ocular melanomas (UVM)</span>
<span class="co">#&gt; datasets() (2 total):</span>
<span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span>
<span class="co">#&gt;   TCGA.UVM.sampleMap/UVM_clinicalMatrix</span>

<span class="co"># only return datasets information</span>
tcga_data<span class="op">$</span>DataInfo
<span class="co">#&gt;                   XenaHosts XenaHostNames                     XenaCohorts</span>
<span class="co">#&gt; 1 https://tcga.xenahubs.net          TCGA TCGA Lung Adenocarcinoma (LUAD)</span>
<span class="co">#&gt; 2 https://tcga.xenahubs.net          TCGA     TCGA Ocular melanomas (UVM)</span>
<span class="co">#&gt;                              XenaDatasets ProjectID  DataType</span>
<span class="co">#&gt; 1 TCGA.LUAD.sampleMap/LUAD_clinicalMatrix      LUAD Phenotype</span>
<span class="co">#&gt; 2   TCGA.UVM.sampleMap/UVM_clinicalMatrix       UVM Phenotype</span>
<span class="co">#&gt;               FileType</span>
<span class="co">#&gt; 1 Clinical Information</span>
<span class="co">#&gt; 2 Clinical Information</span></code></pre></div>
<p>Set <code>download=TRUE</code> to download data, default data will be downloaded to system temp directory (you can specify the path with <code>destdir</code> option):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># only download clinical data</span>
<span class="kw"><a href="../reference/getTCGAdata.html">getTCGAdata</a></span>(<span class="kw">c</span>(<span class="st">"UVM"</span>, <span class="st">"LUAD"</span>), <span class="dt">download =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; We will download files to directory /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//Rtmpw9sHFT.</span>
<span class="co">#&gt; /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//Rtmpw9sHFT/TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz, the file has been download!</span>
<span class="co">#&gt; Downloading TCGA.UVM.sampleMap__UVM_clinicalMatrix.gz</span>
<span class="co">#&gt; Note fileNames transfromed from datasets name and / chracter all changed to __ character.</span></code></pre></div>
<div id="support-data-type-and-options" class="section level4">
<h4 class="hasAnchor">
<a href="#support-data-type-and-options" class="anchor"></a>Support Data Type and Options</h4>
<ul>
<li>clinical information: <code>clinical</code>
</li>
<li>mRNA Sequencing: <code>mRNASeq</code>
</li>
<li>mRNA microarray: <code>mRNAArray</code>
</li>
<li>miRNA Sequencing: <code>miRNASeq</code>
</li>
<li>exon Sequencing: <code>exonRNASeq</code>
</li>
<li>RPPA array: <code>RPPAArray</code>
</li>
<li>DNA Methylation: <code>Methylation</code>
</li>
<li>Gene mutation: <code>GeneMutation</code>
</li>
<li>Somatic mutation: <code>SomaticMutation</code>
</li>
<li>Gistic2 Copy Number: <code>GisticCopyNumber</code>
</li>
<li>Copy Number Segment: <code>CopyNumberSegment</code>
</li>
</ul>
<blockquote>
<p>other data type supported by Xena cannot download use this function. Please refer to <code>downloadTCGA</code> function or <code>XenaGenerate</code> function.</p>
</blockquote>
<p><strong>NOTE</strong>: Sequencing data are all based on Illumina Hiseq platform, other platform (Illumina GA) data supported by Xena cannot download using this function. This is for building consistent data download flow. Mutation use <code>broad automated</code> version (except <code>PANCAN</code> use <code>MC3 Public Version</code>). If you wan to download other datasets, please refer to <code>downloadTCGA</code> function or <code>XenaGenerate</code> function.</p>
</div>
</div>
<div id="download-any-tcga-data-by-datatypes-and-filetypes" class="section level3">
<h3 class="hasAnchor">
<a href="#download-any-tcga-data-by-datatypes-and-filetypes" class="anchor"></a>download any TCGA data by datatypes and filetypes</h3>
<p><code>downloadTCGA</code> function can used to download any TCGA data supported by Xena, but in a way different from <code>getTCGAdata</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># download RNASeq data (use UVM as an example)</span>
<span class="kw"><a href="../reference/downloadTCGA.html">downloadTCGA</a></span>(<span class="dt">project =</span> <span class="st">"UVM"</span>,
                 <span class="dt">data_type =</span> <span class="st">"Gene Expression RNASeq"</span>,
                  <span class="dt">file_type =</span> <span class="st">"IlluminaHiSeq RNASeqV2"</span>)</code></pre></div>
<p>See the arguments:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">args</span>(downloadTCGA)
<span class="co">#&gt; function (project = NULL, data_type = NULL, file_type = NULL, </span>
<span class="co">#&gt;     destdir = tempdir(), force = FALSE, ...) </span>
<span class="co">#&gt; NULL</span></code></pre></div>
<p>Except <code>destdir</code> option, you only need to select three arguments for downloading data. Even throught the number is far less than <code>getTCGAdata</code>, it is more complex than the latter.</p>
<p>Before you download data, you need spare some time to figure out what data type and file type available and what your datasets have.</p>
<p><code>availTCGA</code> can return all information you need:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/availTCGA.html">availTCGA</a></span>()
<span class="co">#&gt; Note not all projects have listed data types and file types, you can use showTCGA function to check if exist</span>
<span class="co">#&gt; $ProjectID</span>
<span class="co">#&gt;  [1] "LAML"     "ACC"      "CHOL"     "BLCA"     "BRCA"     "CESC"    </span>
<span class="co">#&gt;  [7] "COADREAD" "COAD"     "UCEC"     "ESCA"     "FPPP"     "GBM"     </span>
<span class="co">#&gt; [13] "HNSC"     "KICH"     "KIRC"     "KIRP"     "DLBC"     "LIHC"    </span>
<span class="co">#&gt; [19] "LGG"      "GBMLGG"   "LUAD"     "LUNG"     "LUSC"     "SKCM"    </span>
<span class="co">#&gt; [25] "MESO"     "UVM"      "OV"       "PANCAN"   "PAAD"     "PCPG"    </span>
<span class="co">#&gt; [31] "PRAD"     "READ"     "SARC"     "STAD"     "TGCT"     "THYM"    </span>
<span class="co">#&gt; [37] "THCA"     "UCS"     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $DataType</span>
<span class="co">#&gt;  [1] "DNA Methylation"                       </span>
<span class="co">#&gt;  [2] "Gene Level Copy Number"                </span>
<span class="co">#&gt;  [3] "Somatic Mutation"                      </span>
<span class="co">#&gt;  [4] "Gene Expression RNASeq"                </span>
<span class="co">#&gt;  [5] "miRNA Mature Strand Expression RNASeq" </span>
<span class="co">#&gt;  [6] "Gene Somatic Non-silent Mutation"      </span>
<span class="co">#&gt;  [7] "Copy Number Segments"                  </span>
<span class="co">#&gt;  [8] "Exon Expression RNASeq"                </span>
<span class="co">#&gt;  [9] "Phenotype"                             </span>
<span class="co">#&gt; [10] "PARADIGM Pathway Activity"             </span>
<span class="co">#&gt; [11] "Protein Expression RPPA"               </span>
<span class="co">#&gt; [12] "Transcription Factor Regulatory Impact"</span>
<span class="co">#&gt; [13] "Gene Expression Array"                 </span>
<span class="co">#&gt; [14] "Signatures"                            </span>
<span class="co">#&gt; [15] "iCluster"                              </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $FileType</span>
<span class="co">#&gt;  [1] "Methylation27K"                            </span>
<span class="co">#&gt;  [2] "Methylation450K"                           </span>
<span class="co">#&gt;  [3] "Gistic2"                                   </span>
<span class="co">#&gt;  [4] "wustl hiseq automated"                     </span>
<span class="co">#&gt;  [5] "IlluminaGA RNASeq"                         </span>
<span class="co">#&gt;  [6] "IlluminaHiSeq RNASeqV2 in percentile rank" </span>
<span class="co">#&gt;  [7] "IlluminaHiSeq RNASeqV2 pancan normalized"  </span>
<span class="co">#&gt;  [8] "IlluminaHiSeq RNASeqV2"                    </span>
<span class="co">#&gt;  [9] "After remove germline cnv"                 </span>
<span class="co">#&gt; [10] "PANCAN AWG analyzed"                       </span>
<span class="co">#&gt; [11] "Clinical Information"                      </span>
<span class="co">#&gt; [12] "wustl automated"                           </span>
<span class="co">#&gt; [13] "Gistic2 thresholded"                       </span>
<span class="co">#&gt; [14] "Before remove germline cnv"                </span>
<span class="co">#&gt; [15] "Use only RNASeq"                           </span>
<span class="co">#&gt; [16] "Use RNASeq plus Copy Number"               </span>
<span class="co">#&gt; [17] "bcm automated"                             </span>
<span class="co">#&gt; [18] "IlluminaHiSeq RNASeq"                      </span>
<span class="co">#&gt; [19] "bcm curated"                               </span>
<span class="co">#&gt; [20] "broad curated"                             </span>
<span class="co">#&gt; [21] "RPPA"                                      </span>
<span class="co">#&gt; [22] "bsgsc automated"                           </span>
<span class="co">#&gt; [23] "broad automated"                           </span>
<span class="co">#&gt; [24] "bcgsc automated"                           </span>
<span class="co">#&gt; [25] "ucsc automated"                            </span>
<span class="co">#&gt; [26] "RABIT Use IlluminaHiSeq RNASeqV2"          </span>
<span class="co">#&gt; [27] "RABIT Use IlluminaHiSeq RNASeq"            </span>
<span class="co">#&gt; [28] "RPPA normalized by RBN"                    </span>
<span class="co">#&gt; [29] "RABIT Use Agilent 244K Microarray"         </span>
<span class="co">#&gt; [30] "wustl curated"                             </span>
<span class="co">#&gt; [31] "Use Microarray plus Copy Number"           </span>
<span class="co">#&gt; [32] "Use only Microarray"                       </span>
<span class="co">#&gt; [33] "Agilent 244K Microarray"                   </span>
<span class="co">#&gt; [34] "IlluminaGA RNASeqV2"                       </span>
<span class="co">#&gt; [35] "bcm SOLiD"                                 </span>
<span class="co">#&gt; [36] "RABIT Use IlluminaGA RNASeqV2"             </span>
<span class="co">#&gt; [37] "RABIT Use IlluminaGA RNASeq"               </span>
<span class="co">#&gt; [38] "RABIT Use Affymetrix U133A Microarray"     </span>
<span class="co">#&gt; [39] "Affymetrix U133A Microarray"               </span>
<span class="co">#&gt; [40] "MethylMix"                                 </span>
<span class="co">#&gt; [41] "bcm SOLiD curated"                         </span>
<span class="co">#&gt; [42] "Gene Expression Subtype"                   </span>
<span class="co">#&gt; [43] "Platform-corrected PANCAN12 dataset"       </span>
<span class="co">#&gt; [44] "Batch effects normalized"                  </span>
<span class="co">#&gt; [45] "MC3 Public Version"                        </span>
<span class="co">#&gt; [46] "TCGA Sample Type and Primary Disease"      </span>
<span class="co">#&gt; [47] "RPPA pancan normalized"                    </span>
<span class="co">#&gt; [48] "Tumor copy number"                         </span>
<span class="co">#&gt; [49] "Genome-wide DNA Damage Footprint HRD Score"</span>
<span class="co">#&gt; [50] "TCGA Molecular Subtype"                    </span>
<span class="co">#&gt; [51] "iCluster cluster assignments"              </span>
<span class="co">#&gt; [52] "iCluster latent variables"                 </span>
<span class="co">#&gt; [53] "RNA based StemnessScore"                   </span>
<span class="co">#&gt; [54] "DNA methylation based StemnessScore"       </span>
<span class="co">#&gt; [55] "Pancan Gene Programs"                      </span>
<span class="co">#&gt; [56] "Immune Model Based Subtype"                </span>
<span class="co">#&gt; [57] "Immune Signature Scores"</span></code></pre></div>
<p>Note not all datasets have these property, <code>showTCGA</code> can help you to check it. It will return all data in TCGA, you can use following code in RStudio and search your data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">View</span>(<span class="kw"><a href="../reference/showTCGA.html">showTCGA</a></span>())</code></pre></div>
<p><strong>OR you can use shiny app provided by <code>UCSCXenaTools</code> to search</strong>.</p>
<p>Run shiny by:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">UCSCXenaTools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/UCSCXenaTools/topics/XenaShiny">XenaShiny</a></span>()</code></pre></div>
<p>Download by shiny is under consideration, I am try learning more about how to operate shiny.</p>
</div>
</div>
<div id="sessioninfo" class="section level2">
<h2 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.5.1 (2018-07-02)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS High Sierra 10.13.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] zh_CN.UTF-8/zh_CN.UTF-8/zh_CN.UTF-8/C/zh_CN.UTF-8/zh_CN.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] bindrcpp_0.2.2      dplyr_0.7.6         UCSCXenaTools_0.2.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_0.12.18         compiler_3.5.1       pillar_1.3.0        </span>
<span class="co">#&gt;  [4] later_0.7.3          bindr_0.1.1          prettydoc_0.2.1     </span>
<span class="co">#&gt;  [7] tools_3.5.1          digest_0.6.16        jsonlite_1.5        </span>
<span class="co">#&gt; [10] evaluate_0.11        memoise_1.1.0        tibble_1.4.2        </span>
<span class="co">#&gt; [13] pkgconfig_2.0.2      rlang_0.2.2          shiny_1.1.0         </span>
<span class="co">#&gt; [16] rstudioapi_0.7       commonmark_1.5       curl_3.2            </span>
<span class="co">#&gt; [19] yaml_2.2.0           pkgdown_1.1.0        httr_1.3.1          </span>
<span class="co">#&gt; [22] stringr_1.3.1        roxygen2_6.1.0       xml2_1.2.0          </span>
<span class="co">#&gt; [25] knitr_1.20           hms_0.4.2            desc_1.2.0          </span>
<span class="co">#&gt; [28] fs_1.2.6             shinydashboard_0.7.0 rprojroot_1.3-2     </span>
<span class="co">#&gt; [31] tidyselect_0.2.4     glue_1.3.0           R6_2.2.2            </span>
<span class="co">#&gt; [34] rmarkdown_1.10       readr_1.1.1          purrr_0.2.5         </span>
<span class="co">#&gt; [37] magrittr_1.5         backports_1.1.2      promises_1.0.1      </span>
<span class="co">#&gt; [40] htmltools_0.3.6      MASS_7.3-50          assertthat_0.2.0    </span>
<span class="co">#&gt; [43] mime_0.5             xtable_1.8-2         httpuv_1.4.5        </span>
<span class="co">#&gt; [46] stringi_1.2.4        crayon_1.3.4</span></code></pre></div>
</div>
<div id="bug-report" class="section level2">
<h2 class="hasAnchor">
<a href="#bug-report" class="anchor"></a>Bug Report</h2>
<p>I have no time to test if all condition are right and all datasets can normally be downloaded. So if you have any question or suggestion, please open an issue on Github at <a href="https://github.com/ShixiangWang/UCSCXenaTools/issues" class="uri">https://github.com/ShixiangWang/UCSCXenaTools/issues</a>.</p>
</div>
<div id="acknowledgement" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgement" class="anchor"></a>Acknowledgement</h2>
<p>This package is based on <a href="https://github.com/mtmorgan/XenaR">XenaR</a>, thanks <a href="https://github.com/mtmorgan">Martin Morgan</a> for his work.</p>
</div>
<div id="license" class="section level2">
<h2 class="hasAnchor">
<a href="#license" class="anchor"></a>LICENSE</h2>
<p>GPL-3</p>
<p>please note, code from XenaR package under Apache 2.0 license.</p>
</div>
<div id="todo" class="section level2">
<h2 class="hasAnchor">
<a href="#todo" class="anchor"></a>ToDo</h2>
<ul>
<li>Shinny and more</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#data-hub-list">Data Hub List</a></li>
      <li><a href="#usage">Usage</a></li>
      <li><a href="#tcga-common-data-easy-download">TCGA Common Data Easy Download</a></li>
      <li><a href="#sessioninfo">SessionInfo</a></li>
      <li><a href="#bug-report">Bug Report</a></li>
      <li><a href="#acknowledgement">Acknowledgement</a></li>
      <li><a href="#license">LICENSE</a></li>
      <li><a href="#todo">ToDo</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Shixiang Wang, Martin Morgan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
