<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get TCGA Common Data Sets by Project ID and Property — getTCGAdata • UCSCXenaTools</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Get TCGA Common Data Sets by Project ID and Property — getTCGAdata" />

<meta property="og:description" content="This is the most useful function for user to download common
TCGA datasets, it is similar to getFirehoseData function in RTCGAToolbox
package." />

<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">UCSCXenaTools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/USCSXenaTools.html">UCSCXenaTools: Download Public Cancer Genomic Data from UCSC Xena Hubs</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ShixiangWang/UCSCXenaTools">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get TCGA Common Data Sets by Project ID and Property</h1>
    <small class="dont-index">Source: <a href='https://github.com/ShixiangWang/UCSCXenaTools/blob/master/R/simplify.R'><code>R/simplify.R</code></a></small>
    <div class="hidden name"><code>getTCGAdata.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This is the most useful function for user to download common
TCGA datasets, it is similar to <code>getFirehoseData</code> function in <code>RTCGAToolbox</code>
package.</p>
    
    </div>

    <pre class="usage"><span class='fu'>getTCGAdata</span>(<span class='kw'>project</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>clinical</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>download</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>forceDownload</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>destdir</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/tempfile'>tempdir</a></span>(), <span class='kw'>mRNASeq</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>mRNAArray</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>mRNASeqType</span> <span class='kw'>=</span> <span class='st'>"normalized"</span>, <span class='kw'>miRNASeq</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>exonRNASeq</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>RPPAArray</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>ReplicateBaseNormalization</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>Methylation</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>MethylationType</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"27K"</span>, <span class='st'>"450K"</span>), <span class='kw'>GeneMutation</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>SomaticMutation</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>GisticCopyNumber</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>Gistic2Threshold</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>CopyNumberSegment</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>RemoveGermlineCNV</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>project</th>
      <td><p>default is <code>NULL</code>. Should be one or more of TCGA project id (character vector) provided by Xena.
See all available project id, please use <code><a href='availTCGA.html'>availTCGA("ProjectID")</a></code>.</p></td>
    </tr>
    <tr>
      <th>clinical</th>
      <td><p>logical. if <code>TRUE</code>, download clinical information. Default is <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>download</th>
      <td><p>logical. if <code>TRUE</code>, download data, otherwise return a result list include data
information. Default is <code>FALSE</code>. You can set this to <code>FALSE</code> if you want to check what you will download or
use other function provided by <code>UCSCXenaTools</code> to filter result datasets you want to download.</p></td>
    </tr>
    <tr>
      <th>forceDownload</th>
      <td><p>logical. if <code>TRUE</code>, force to download files no matter if exist. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>destdir</th>
      <td><p>specify a location to store download data. Default is system temp directory.</p></td>
    </tr>
    <tr>
      <th>mRNASeq</th>
      <td><p>logical. if <code>TRUE</code>, download mRNASeq data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>mRNAArray</th>
      <td><p>logical. if <code>TRUE</code>, download mRNA microarray data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>mRNASeqType</th>
      <td><p>character vector. Can be one, two or three
in <code><a href='https://www.rdocumentation.org/packages/base/topics/c'>c("normalized", "pancan normalized", "percentile")</a></code>.</p></td>
    </tr>
    <tr>
      <th>miRNASeq</th>
      <td><p>logical. if <code>TRUE</code>, download miRNASeq data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>exonRNASeq</th>
      <td><p>logical. if <code>TRUE</code>, download exon RNASeq data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>RPPAArray</th>
      <td><p>logical. if <code>TRUE</code>, download RPPA data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>ReplicateBaseNormalization</th>
      <td><p>logical. if <code>TRUE</code>, download RPPA data by Replicate Base
Normalization (RBN). Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>Methylation</th>
      <td><p>logical. if <code>TRUE</code>, download DNA Methylation data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>MethylationType</th>
      <td><p>character vector. Can be one or two in <code><a href='https://www.rdocumentation.org/packages/base/topics/c'>c("27K", "450K")</a></code>.</p></td>
    </tr>
    <tr>
      <th>GeneMutation</th>
      <td><p>logical. if <code>TRUE</code>, download gene mutation data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>SomaticMutation</th>
      <td><p>logical. if <code>TRUE</code>, download somatic mutation data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>GisticCopyNumber</th>
      <td><p>logical. if <code>TRUE</code>, download Gistic2 Copy Number data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>Gistic2Threshold</th>
      <td><p>logical. if <code>TRUE</code>, download Threshold Gistic2 data. Default is <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>CopyNumberSegment</th>
      <td><p>logical. if <code>TRUE</code>, download Copy Number Segment data. Default is <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>RemoveGermlineCNV</th>
      <td><p>logical. if <code>TRUE</code>, download Copy Number Segment data which has removed
germline copy number variation. Default is <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>other argument to <code>download.file</code> function</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>if <code>download=TRUE</code>, return <code>data.frame</code> from <code>XenaDownload</code>,
otherwise return a list including <code>XenaHub</code> object and datasets information</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>TCGA Common Data Sets are frequently used for biological analysis.
To make easier to achieve these data, this function provide really easy
options to choose datasets and behavior. All availble information about
datasets of TCGA can access vis <code><a href='availTCGA.html'>availTCGA()</a></code> and check with <code><a href='showTCGA.html'>showTCGA()</a></code>.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='co'># NOT RUN {</span>
<span class='co'>###### get data, but not download</span>

<span class='co'># 1 choose project and data types you wanna download</span>
<span class='fu'>getTCGAdata</span>(<span class='kw'>project</span> <span class='kw'>=</span> <span class='st'>"LUAD"</span>, <span class='kw'>mRNASeq</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>mRNAArray</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
<span class='kw'>mRNASeqType</span> <span class='kw'>=</span> <span class='st'>"normalized"</span>, <span class='kw'>miRNASeq</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>exonRNASeq</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
<span class='kw'>RPPAArray</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>Methylation</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>MethylationType</span> <span class='kw'>=</span> <span class='st'>"450K"</span>,
<span class='kw'>GeneMutation</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>SomaticMutation</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)

<span class='co'># 2 only choose 'LUAD' and its clinical data</span>
<span class='fu'>getTCGAdata</span>(<span class='kw'>project</span> <span class='kw'>=</span> <span class='st'>"LUAD"</span>)
<span class='co'># }</span><span class='co'># NOT RUN {</span>
<span class='co'>###### download datasets</span>

<span class='co'># 3 download clinical datasets of LUAD and LUSC</span>
<span class='fu'>getTCGAdata</span>(<span class='kw'>project</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"LUAD"</span>, <span class='st'>"LUSC"</span>), <span class='kw'>clinical</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>download</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)

<span class='co'># 4 download clinical, RPPA and gene mutation datasets of LUAD and LUSC</span>
<span class='co'># getTCGAdata(project = c("LUAD", "LUSC"), clinical = TRUE, RPPAArray = TRUE, GeneMutation = TRUE)</span>
<span class='co'># }</span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Shixiang Wang <a href='mailto:w_shixiang@163.com'>w_shixiang@163.com</a></p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Shixiang Wang, Martin Morgan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

